implementations/rust/ockam/ockam_api/src/logs/grpc_forwarder.rs

```diff
@@ -0,0 +1,84 @@
+use crate::logs::secure_client_service::OckamGrpcRequest;
+use crate::{ApiError, Result};
+use hyper::{http, Uri};
+use ockam_core::api::{Method, Request};
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::{async_trait, Routed, Worker};
+use ockam_node::Context;
+use std::future;
+use tonic::body::BoxBody;
+use tonic::client::GrpcService;
+use tonic::transport::Channel;
+
+/// The GrpcForwarder worker accepts gRPC requests serialized as Ockam messages
+/// and forwards them to an HTTP endpoint.
+///
+/// Note that we don't wait for a response from the endpoint.
+pub struct GrpcForwarder {
+    channel: Channel,
+}
+
+impl GrpcForwarder {
+    /// Create a Channel for the given URI
+    pub async fn new(uri: Uri) -> Result<Self> {
+        let channel = Channel::builder(uri.clone())
+            .tls_config(tonic::transport::ClientTlsConfig::new().with_native_roots())
+            .map_err(|e| {
+                ApiError::message(format!(
+                    "cannot create a TLS config for the HttpForwarder channel at {uri:?}: {e:?}"
+                ))
+            })?
+            .connect()
+            .await
+            .map_err(|e| ApiError::message(format!("cannot connect to {uri:?}: {e:?}")))?;
+
+        Ok(Self { channel })
+    }
+
+    /// Forward an http Request.
+    /// We don't wait for a response here.
+    async fn forward_http_request(&mut self, request: http::Request<BoxBody>) -> Result<()> {
+        self.ready().await.map_err(ApiError::core)?;
```

Comment:
Why is this needed before forwarding every request?

Reply:
This is the protocol used by the `tonic` library when calling a `Channel`.